packages/wonder-blocks-modal/components/two-column-modal.js

```diff
@@ -35,33 +42,73 @@ type Props = {
 export default class TwoColumnModal extends React.Component<Props> {
     static defaultProps = {
         onClickCloseButton: () => {},
+        smallSidebar: false,
     };
 
     render() {
+        const {sidebar, footer, contents, smallSidebar} = this.props;
+
+        const sidebarView = smallSidebar ? (
+            <Cell
+                smCols={4}
+                largeCols={1}
+                style={[styles.column, styles.bgSidebar, styles.bgSidebarSmall]}
+            >
+                {sidebar}
+            </Cell>
+        ) : (
+            <FlexCell style={[styles.column, styles.bgSidebar]}>
+                <View style={styles.sidebar}>{sidebar}</View>
+            </FlexCell>
+        );
+
```

Comment:
I wonder if maybe we could pull all this stuff into a separate `_renderSidebar` method? Then use `if` blocks and `return` statements to handle the `null`, `smallSidebar`, and `!smallSidebar` cases? In this form, it's hard for me to keep track of the flow of rendering.

Reply:
Good call! I can do that.